nupic/research/frameworks/dendrites/metrics/metrics.py

```diff
@@ -199,6 +199,48 @@ def hidden_activations_by_unit(activations, targets):
         return habu
 
 
+def contexts_by_class(contexts, targets):
+    """
+    Returns a 2D torch tensor with shape (num_categories, dim_context) where cell c, k
+    gives the fraction of instances of category c for which feature/dimension k of the
+    context signal generated for an input of said class was non-zero. All values are in
+    the range [0, 1].
+
+    :param activations: 2D torch tensor with shape (batch_size, num_units) where entry
+                        b, i gives the activation of unit i for example b
+    :param targets: 1D torch tensor with shape (batch_size,) where entry b gives the
+                    target label for example b
+    """
+
+    with torch.no_grad():
+
+        device = contexts.device
+
+        # Assume the following:
+        # - target values are zero-based
+        # - the largest target value in the batch is that amongst all data
+        num_categories = 1 + targets.max().item()
+        _, dim_context = contexts.size()
+
+        _contexts_by_class = torch.zeros((0, dim_context))
```

Comment:
My only reason is because the function has the same name (i.e., `context_by_class`), so I wanted to distinguish it. I guess I could just call it `cbc` or something of that sort.

Reply:
Fixed.